velox/functions/sparksql/RegisterCompare.cpp

```diff
@@ -16,18 +16,28 @@
 #include "velox/functions/sparksql/RegisterCompare.h"
 
 #include "velox/functions/lib/RegistrationHelpers.h"
-#include "velox/functions/prestosql/Comparisons.h"
+#include "velox/functions/sparksql/CompareFunctionsNullSafe.h"
+#include "velox/functions/sparksql/Comparisons.h"
 
 namespace facebook::velox::functions::sparksql {
 
 void registerCompareFunctions(const std::string& prefix) {
-  registerBinaryScalar<EqFunction, bool>({prefix + "equalto"});
-  registerBinaryScalar<NeqFunction, bool>({prefix + "notequalto"});
-  registerBinaryScalar<LtFunction, bool>({prefix + "lessthan"});
-  registerBinaryScalar<GtFunction, bool>({prefix + "greaterthan"});
-  registerBinaryScalar<LteFunction, bool>({prefix + "lessthanorequal"});
-  registerBinaryScalar<GteFunction, bool>({prefix + "greaterthanorequal"});
-
+  // Register compare functions
+  exec::registerStatefulVectorFunction(
```

Comment:
Can it still be SimpleFunction?

Reply:
Looks like it is stateless https://github.com/facebookincubator/velox/pull/4029#discussion_r1130514350

Reply:
These functions are stateless. It has some problems when try to use `registerVectorFunction` API as there are two input parameters needed as input when constructing the function, which will cause convert failure.  It also happens for the existing "EqualNullSafe" and should be why it also use the `registerStatefulVectorFunction`. By the way, `registerVectorFunction` actually do a function factory wrapper and then call `registerStatefulVectorFunction` eventually.

Reply:
@Yuhta Any suggestion on this open? I see Presto has SIMD implementation for these functions. Maybe we can have such for Spark in later PR.

Reply:
Is there any difference between the comparisons in Presto and Spark?  Maybe we should move the one in Presto to a common place and reuse them in both engines.

Reply:
Yes. Spark comparison functions have some different rule for `NaN` handling, so they will call `Cmp` like [Less](https://github.com/facebookincubator/velox/blob/d18d67f966dbdbddffb8f3fd6ed0e4e0c8ebc2e0/velox/functions/sparksql/Comparisons.h#L34-L39C6).